src/huggingface_hub/hub_mixin.py

```diff
@@ -406,44 +403,48 @@ def push_to_hub(
 
 
 class PyTorchModelHubMixin(ModelHubMixin):
-    def __init__(self, *args, **kwargs):
-        """
-        Mix this class with your torch-model class for ease process of saving &
-        loading from huggingface-hub.
-
-        Example:
-
-        ```python
-        >>> from huggingface_hub import PyTorchModelHubMixin
-
-
-        >>> class MyModel(nn.Module, PyTorchModelHubMixin):
-        ...     def __init__(self, **kwargs):
-        ...         super().__init__()
-        ...         self.config = kwargs.pop("config", None)
-        ...         self.layer = ...
-
-        ...     def forward(self, *args):
-        ...         return ...
-
-
-        >>> model = MyModel()
-        >>> model.save_pretrained(
-        ...     "mymodel", push_to_hub=False
-        >>> )  # Saving model weights in the directory
-        >>> model.push_to_hub(
-        ...     repo_id="mymodel", commit_message="model-1"
-        >>> )  # Pushing model-weights to hf-hub
+    """
+    Implementation of [`ModelHubMixin`] to provide model Hub upload/download
+    capabilities to PyTorch models. The model is set in evaluation mode by
+    default using `model.eval()` (dropout modules are deactivated). To train
+    the model, you should first set it back in training mode with
+    `model.train()`.
+
+    Example:
+
+    ```python
+    >>> import torch
+    >>> import torch.nn as nn
+    >>> from huggingface_hub import PyTorchModelHubMixin
+
+
+    >>> class MyModel(nn.Module, PyTorchModelHubMixin):
+    ...     def __init__(self):
+    ...         super().__init__()
+    ...         self.param = nn.Parameter(torch.rand(3, 4))
+    ...         self.linear = nn.Linear(4, 5)
+
+    ...     def forward(self, x):
+    ...         return self.linear(x + self.param)
+
+
+    >>> model = MyModel()
+    >>> # Save model weights to local directory
+    >>> model.save_pretrained("my-awesome-model")
+    >>> # Push model weights to the Hub
+    >>> model.push_to_hub("my-awesome-model")
+    >>> # Download and initialize weights from the Hub
+    >>> model = MyModel.from_pretrained("username/my-awesome-model")
+    ```
+    """
 
-        >>> # Downloading weights from hf-hub & model will be initialized from those weights
-        >>> model = MyModel.from_pretrained("username/mymodel@main")
-        ```
-        """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
This is now useless I think as Python would automatically take the parent one is `__init__` is not defined.

Reply:
Good catch! Done in https://github.com/huggingface/huggingface_hub/pull/1139/commits/566c3f321bcd1d990d9415bc7ce48ed0689ce98b